cpp/csp/adapters/websockets/ClientOutputAdapter.h

```diff
@@ -0,0 +1,35 @@
+#ifndef _IN_CSP_ADAPTERS_WEBSOCKETS_CLIENT_OUTPUTADAPTER_H
+#define _IN_CSP_ADAPTERS_WEBSOCKETS_CLIENT_OUTPUTADAPTER_H
+
+#include <csp/adapters/websockets/WebsocketEndpoint.h>
+#include <csp/engine/Dictionary.h>
+#include <csp/engine/OutputAdapter.h>
+#include <csp/adapters/utils/MessageWriter.h>
+
+namespace csp::adapters::websockets
+{
+
+
+class ClientOutputAdapter final: public csp::OutputAdapter
+{
+
+public:
+    ClientOutputAdapter(
+        csp::Engine * engine,
+        WebsocketEndpointBase* endpoint
+    );
+
+    void executeImpl() override;
+
+    const char * name() const override { return "ClientOutputAdapter"; }
+
+private:
+    WebsocketEndpointBase* m_endpoint;
+    websocketpp::connection_hdl* m_hdl;
```

Comment:
As a general rule I think it's better to include headers where they are used rather than to let them trickle in from other includes, so let's include the relevant `websocketpp` header here. Two reasons for this:

1) If dependencies change in another file it won't affect this file
2) Easier to find where the code is coming from if the include path is present